### DISCUSSION TITLE
Pull request can be merged after commits after approval.

### DISCUSSION THREAD

**Discussion by GokhanBond** [community/community#37090] [Pull Requests] [answered]
Hello!

First of all, it is great to see there is a community answering and helping each other.
Last week I have found a unexpected flow whilst pull requesting/merging. Let me explain the situation and what I find weird.

1. I have created a pull request from <branch> to <development>. The development has a prerequisite to have at least 1 approval.
2. A colleague has approved my beautiful changes. but no merge has been done yet.
3. I found a small mistake in the code and could change it directly within GitHub.
4. The pull request was still mergable.

So my question is, why can I still merge it? I did change the code **after** the approval.
I truly did expect it would 'undo' approval.

Can someone explain this to me?
Thanks in advance 😄

**Comment by MitkoVtori** [comment, marked as answer]
Hi, the problem is in the branch protectinon rules. If you're the owner, just go and add this rule, if you're not, tell someone with permissions to change this:
![image](https://user-images.githubusercontent.com/112943652/197485339-4164040e-146d-44e5-8998-c758ddd52494.png)
This should be enabled so when you push new changes you won't be able to merge until someone approve again.

**Reply by GokhanBond** [reply]
Ah, I see. The IT-guy is the owner of the repositories, and I have already asked for permission, or to change the rules, as you have showed.
Thanks for your amazing reply! It is super helpful!

**Comment by ossanna16** [comment]
Hi there @GokhanBond and welcome to our community! Thank you for asking a great question 🙂
